engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -198,11 +205,19 @@ static Object doCached(
         @Cached(value = "member") String cachedMember,
         @Cached(value = "buildSym(cachedConstructor, cachedMember)") UnresolvedSymbol cachedSym,
         @CachedLibrary("cachedSym") InteropLibrary symbols)
-        throws UnsupportedMessageException, ArityException, UnsupportedTypeException {
+        throws UnsupportedMessageException, ArityException, UnsupportedTypeException, UnknownIdentifierException {
       Object[] args = new Object[arguments.length + 1];
       args[0] = receiver;
       System.arraycopy(arguments, 0, args, 1, arguments.length);
-      return symbols.execute(cachedSym, args);
+      try {
+        return symbols.execute(cachedSym, args);
+      } catch (PanicException ex) {
+        if (ex.getCause() instanceof UnknownIdentifierException interopEx) {
+          throw interopEx;
+        } else {
+          throw ex;
+        }
```

Comment:
I think I don't fully understand why this piece of logic is needed here in particular?

Instead of 'converting' the exception here, couldn't we just at the throw-site, throw a different exception depending on the `onBoundary` value?

Or would it cause problems because the exception could leak "too much"?

-----

I guess my question is, is every `onBoundary=true` call, coupled with this particular place where we convert the exception, or if there are other places where it can be called.

And if the latter - is it deliberate that in these other places, the conversion _does not_ take place?

Reply:
> couldn't we just at the throw-site, throw a different exception depending on the onBoundary value?

I tried that, but it doesn't compile easily. The exception is constructed in `execute` method - that method cannot throw `UnknownIdentifierException` only `invokeMember` can throw such exception. As such I decided to wrap&extract the exception.

Reply:
> is it deliberate that in these other places, the conversion does not take place

If the conversion isn't done, then `PanicException` is thrown - there is nothing wrong on yielding `PanicException`s.

Reply:
Ok, thank you for the explanation.